docs/docs/typescript/utility-types.md

```diff
@@ -11,6 +11,21 @@ You'll see these helpers quite often if you use the generators, so let's walk th
 
 Cells created using the generators come with all the types your normally need, including the `CellSuccessProps`, `CellFailureProps`, and `CellLoadingProps` utility types.
 
+### `BeforeQueryHook<CellProps, CellVariables>`
+
+This is used to type the `beforeQuery` function with the prop types and return types.
+
+```ts title="web/src/component/BlogPostCell.tsx"
+import type { BeforeQueryHook } from "@redwoodjs/web"
+
+export const beforeQuery: BeforeQueryHook = (props) => {
```

Comment:
When I tried this type out in a project, it complained about needing two generics:

<img width="916" alt="image" src="https://github.com/redwoodjs/redwood/assets/32992335/9b8f2fbc-8469-4883-bd96-76479ba9e65e">